### PR TITLE
binance_cfutures: TimeOnForce is an optional order submission field

### PR DESCRIPTION
**Pull request by lrascao**
Do not add it to the request unless filled out as the API will deny the
request otherwise.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

- [X] go test ./... -race
- [X] golangci-lint run
- [ ] Test X

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/binance/binance_wrapper.go [hunk 1]
```diff
@@ -975,13 +975,19 @@ func (b *Binance) SubmitOrder(ctx context.Context, s *order.Submit) (order.Submi
 		default:
 			return submitOrderResponse, errors.New("invalid type, check api docs for updates")
 		}
+
+		var timeInForce string
+		if s.Type == order.Limit {
+			timeInForce = "GCT"
```
Comment (gloriousCode) [inline exchanges/binance/binance_wrapper.go:981]: This can be placed in the switch case above

File: exchanges/binance/binance_cfutures.go [hunk 2]
```diff
@@ -1000,7 +1000,10 @@ func (b *Binance) FuturesNewOrder(ctx context.Context, x *FuturesNewOrderRequest
 		params.Set("positionSide", x.PositionSide)
 	}
 	params.Set("type", x.OrderType)
-	params.Set("timeInForce", x.TimeInForce)
+	// TimeInForce is an optional parameter, do not add it unless requested
```
Comment (gloriousCode) [inline exchanges/binance/binance_cfutures.go:1003]: No need for the comment

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for this! Change requests are really minor

File: exchanges/binance/binance_wrapper.go [hunk 3]
```diff
@@ -956,10 +956,15 @@ func (b *Binance) SubmitOrder(ctx context.Context, s *order.Submit) (order.Submi
...
-		var oType string
+		var (
+			oType       string
+			timeInForce string
+		)
+
 		switch s.Type {
 		case order.Limit:
 			oType = cfuturesLimit
+			timeInForce = "GCT"
```
Comment (shazbert) [inline exchanges/binance/binance_wrapper.go:967]: Eyeballing this seems like it needs to change from GCT -> GTC

**Review by shazbert — CHANGES_REQUESTED** [review]
Just one thing from me 🚀

File: exchanges/binance/binance_wrapper.go (hunk 3 above)
Comment (lrascao) [reply exchanges/binance/binance_wrapper.go:967]: wow, that's a Freudian slip right there

Comment (shazbert) [reply exchanges/binance/binance_wrapper.go:967]: 🤣

File: exchanges/binance/binance_wrapper.go [hunk 4]
```diff
@@ -956,10 +956,15 @@ func (b *Binance) SubmitOrder(ctx context.Context, s *order.Submit) (order.Submi
...
-		var oType string
+		var (
+			oType       string
+			timeInForce string
+		)
+
 		switch s.Type {
 		case order.Limit:
 			oType = cfuturesLimit
+			timeInForce = "GTC"
```
Comment (thrasher-) [inline exchanges/binance/binance_wrapper.go:967]: Can use our built-in time in force types:

```suggestion
			timeInForce = BinanceRequestParamsTimeGTC
```

Then this looks good to go! 

image

**Review by thrasher- — CHANGES_REQUESTED** [review]

File: exchanges/binance/binance_wrapper.go (hunk 4 above)
Comment (gloriousCode) [reply exchanges/binance/binance_wrapper.go:967]: Only thing to add is that in order to support this, you'll have to use `string(x)` at some point, which I'd suggest be inside `FuturesNewOrder` to support/use of  `RequestParamsTimeForceType` type
eg
```
if x.TimeInForce != "" {
		params.Set("timeInForce", string(x.TimeInForce))
	}
```
Like in `newOrder`

**Review by gloriousCode — COMMENTED** [review]
Thanks for making those changes!

File: exchanges/binance/binance_wrapper.go (hunk 4 above)
Comment (lrascao) [reply exchanges/binance/binance_wrapper.go]: sounds good, also ended up changing `FuturesNewOrderRequest` `TimeInForce` field to the `RequestParamsTimeForceType` type

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes

**Review by thrasher- — APPROVED** [review]
Thanks @lrascao !

**Review by shazbert — APPROVED** [review]
ACK. Thanks for the changes!

**Pull request merged**